src/geometry/cpu/geometry_op_impl.cc

```diff
@@ -0,0 +1,100 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file array/cpu/geometry_op_impl.cc
+ * \brief Geometry operator CPU implementation
+ */
+#include <dgl/array.h>
+#include <numeric>
+#include <vector>
+
+namespace dgl {
+using runtime::NDArray;
+namespace geometry {
+namespace impl {
+
+/*!
+ * \brief Farthest Point Sampler without the need to compute all pairs of distance.
+ * 
+ * The input array has shape (N, d), where N is the number of points, and d is the dimension.
+ * It consists of a (flatten) batch of point clouds.
+ *
+ * In each batch, the algorithm starts with the sample index specified by ``start_idx``.
+ * Then for each point, we maintain the minimum to-sample distance.
+ * Finally, we pick the point with the maximum such distance.
+ * This process will be repeated for ``sample_points`` - 1 times.
+ */
+template <DLDeviceType XPU, typename FloatType, typename IdType>
+void FarthestPointSampler(NDArray array, int64_t batch_size, int64_t sample_points,
```

Comment:
Have a check of the validity of dist, start_idx and result.

Reply:
Is there a standard MACRO for such check?

Reply:
`CHECK(cond) << msg`, `CHECK_LE(a, b) << msg`, `CHECK_EQ(a, b) << msg`, etc.